### PR TITLE
[@types/httptoolkit__esm] added types for @httptoolkit/esm

### PR DESCRIPTION
**Pull request by nktnet1**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `pnpm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

If adding a new definition:

- [x] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [x] If this is for an npm package, match the name. If not, do not conflict with the name of an npm package.
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [x] `tsconfig.json` [should have](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#tsconfigjson) `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

---

I'm adding types for https://github.com/httptoolkit/esm, an active fork of the https://github.com/standard-things/esm library. There are 2 things I am unsure about:

1. Since the API is identical, I added [@types/esm](https://www.npmjs.com/package/@types/esm) as a dependencies to `package.json` and has the following in `index.d.ts`:
    ```ts
    import esm = require("esm");
    export = esm;
    ```
    Is this the correct approach?

2. The library is named  `@httptoolkit/esm`, which resulted in the generated directory name being `httptoolkit__esm` (and thus the test file is named `httptoolkit__esm-test.ts`. Is this acceptable?

Thank you!

### CODE REVIEW

**Comment by jakebailey** [comment]
Given the main esm package is archived, I would probably copy its types instead of depending on them. The fork may evolve and the types may eventually get removed from the repo.

**Comment by nktnet1** [comment]
Hi @jakebailey,

Thank you for your review. I have copied the types from the original @types/esm below
- https://github.com/DefinitelyTyped/DefinitelyTyped/blob/104726b404e1c7942cb6c895d90002e3eb3a8020/types/esm/index.d.ts

Please let me know if there are any other things I should update :)

File: types/httptoolkit__esm/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,30 @@
+// Project: https://github.com/httptoolkit/esm
```
Comment (jakebailey) [inline types/httptoolkit__esm/index.d.ts:1]: Please remove this header

Comment (nktnet1) [reply types/httptoolkit__esm/index.d.ts]: Hi @jakebailey,

I've removed the `// Project ...` line - did you also want the line after that removed?

Comment (jakebailey) [reply types/httptoolkit__esm/index.d.ts]: Yes, all of it.

Comment (nktnet1) [reply types/httptoolkit__esm/index.d.ts]: Done :)

**Review by jakebailey — APPROVED** [review]

**Pull request merged**